service/history/execution/context.go

```diff
@@ -852,6 +854,7 @@ func (c *contextImpl) UpdateWorkflowExecutionWithNew(
 		domainName,
 		resp.MutableStateUpdateSessionStats,
 	)
+	c.emitLargeWorkflowShardIDStats(currentWorkflowSize-oldWorkflowSize, oldWorkflowHistoryCount, oldWorkflowSize)
```

Comment:
while I'm exploring the code in here:
would any of these be more-correct if they came from `newWorkflow`?  and/or would using `newWorkflow` mean we wouldn't need to maintain as much separately?

I'm not quite sure what the difference is tbh.  I'd have to read more carefully.  I don't think it'll be _dangerous_ to use the wrong one, just possibly misleading (due to subtly-wrong values, or due to unnecessary duplicate calculations that could drift)

Reply:
I don't think it matters. From what I can tell new workflow kind of derives the numbers the same way I am. Maybe it would be less duplicated code but at most it's saving 5 lines and makes it a bit more confusing to read imo.

Reply:
the main reason to prefer using the new-workflow's data is that it'd make it clear what the source of truth is, and get rid of any risk of the calculations drifting away from that source of truth.

which is an "if".  tbh I'm not sure if it's more-correct or not.

Reply:
I could try testing it later? The other stuff is kinda calculated the same way so i wanted to make it consistent. I don't wanna just change part of it then it's super confusing if we never change the rest.

Reply:
which ones are those?
but yea, later's fine (if ever)